### PR TITLE
[ES-1100] Integration with API - v2 kyc-auth endpoint to give out claim verification details

### PR DESCRIPTION
**Pull request by Md-Humair-KK**
Integration with API - v2 kyc-auth endpoint to give out claim verification details

### CODE REVIEW

File: mock-plugin/src/main/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImpl.java [hunk 1]
```diff
@@ -87,8 +87,8 @@ public void validate(String action, ProfileDto profileDto) throws InvalidProfile
         JsonNode inputJson = profileDto.getIdentity();
         List<String> requiredFields = action.equals("CREATE") ? requiredFieldsOnCreate : requiredFieldsOnUpdate;
         for (String fieldName : requiredFields) {
-            if (!inputJson.hasNonNull(fieldName) || (inputJson.get(fieldName).isArray() && inputJson.get(fieldName).isEmpty())) {
-                log.error("Null value found in the required field of {}, required: {}", fieldName, requiredFieldsOnCreate);
+            if (!fieldName.isEmpty() && (!inputJson.hasNonNull(fieldName) || (inputJson.get(fieldName).isArray() && inputJson.get(fieldName).isEmpty()))) {
```
Comment (ase-101) [inline mock-plugin/src/main/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImpl.java:90]: This change already exists in the current 'develop' branch. 
FYR https://github.com/mosip/esignet-plugins/blob/develop/mock-plugin/src/main/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImpl.java#L90

was the latest update taken before raising the PR? @Md-Humair-KK

File: mock-plugin/src/main/resources/mock-idv-user-story.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "scenes":[
-    {"frameNumber" : 0, "stepCode" : "START", "step" : { "code" : "liveness_check", "framesPerSecond" : 3, "durationInSeconds" : 100, "startupDelayInSeconds" : 2, "retryOnTimeout" : false, "retryableErrorCodes" : [] }, "feedback" : null },
+    {"frameNumber" : 0, "stepCode" : "START", "step" : { "code" : "liveness_check", "framesPerSecond" : 1, "durationInSeconds" : 100, "startupDelayInSeconds" : 2, "retryOnTimeout" : false, "retryableErrorCodes" : [] }, "feedback" : null },
```
Comment (ase-101) [inline mock-plugin/src/main/resources/mock-idv-user-story.json:3]: https://github.com/mosip/esignet-plugins/blob/develop/mock-plugin/src/main/resources/mock-idv-user-story.json#L3 This change also exists. Kindly cross-check.

File: mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthRequest.java [hunk 3]
```diff
@@ -29,7 +31,7 @@ public class IdaKycAuthRequest {
     private String requestSessionKey;
     private Map<String, Object> metadata;
     private List<String> allowedKycAttributes;
-
+    private Boolean claimMetadataRequired;
     @Data
```
Comment (ase-101) [inline mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthRequest.java:35]: Please provide on new line gap between L34 and L35

File: mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthResponse.java [hunk 4]
```diff
@@ -5,12 +5,19 @@
...
 
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.JsonNode;
 import lombok.Data;
 
+import java.util.List;
+import java.util.Map;
+
 @Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (ase-101) [inline mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthResponse.java:16]: Why is this annotation required here?

File: mosip-identity-plugin/src/test/java/io/mosip/esignet/plugin/mosipid/service/IdaAuthenticatorImplTest.java
Comment (ase-101) [inline mosip-identity-plugin/src/test/java/io/mosip/esignet/plugin/mosipid/service/IdaAuthenticatorImplTest.java:1]: Lets remove the commented lines, as its no more required.

**Review by ase-101 — CHANGES_REQUESTED** [review]
Please check the comments.

File: mosip-identity-plugin/src/test/java/io/mosip/esignet/plugin/mosipid/service/IdaAuthenticatorImplTest.java
Comment (Md-Humair-KK) [reply mosip-identity-plugin/src/test/java/io/mosip/esignet/plugin/mosipid/service/IdaAuthenticatorImplTest.java:1]: Removed the comment lines of code

File: mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthResponse.java (hunk 4 above)
Comment (Md-Humair-KK) [reply mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthResponse.java]: Referred the TokenResponse DTO and there it was added for a field so added for response as well, but any how the verified claims will be requested based on claimsMetadata flag. Hence removed it from the response DTO.

File: mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthRequest.java (hunk 3 above)
Comment (Md-Humair-KK) [reply mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/dto/IdaKycAuthRequest.java]: Provided gap now

**Pull request closed without merging**